### PR TITLE
we can now infer variable sorts

### PR DESCRIPTION
**Pull request by grosu**
@cos, this is trivial, please merge it.

### CODE REVIEW

File: k-distribution/tutorial/1_k/1_lambda/lesson_5/NOTES.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
 <!-- Copyright (c) 2016 K Team. All Rights Reserved. -->
 
 The builtins have changed, they are now generic for all backends.
+
+Talk about sort inference for variables, for example from `I1 +Int I2`
+we infer the sort of `I1` and `I2` is `Int`.
+
+Check the entire tutorial for instances where we give the sort of a
+variable but we don't have to.  May of those are artifacts since we were
```
Comment (radumereuta) [inline k-distribution/tutorial/1_k/1_lambda/lesson_5/NOTES.md]: Typo: *Many

**Comment by radumereuta** [comment]
If that typo is not important, you can merge.

**Comment by grosu** [comment]
fixed, thanks @radumereuta

**Comment by cos** [comment]
The sort annotations are semantically relevant here. It's nice we can infer them, but I feel we're making the code more cryptic this way. It's probably not be the time for this discussion, just wanted to note it.

**Comment by grosu** [comment]
We don't use the sorts later in lambda++, which is why i modified these lessons, too.  I'll explain this in the README next time I revise the tutorial, probably before the invited tutorial at ETAPS.

**Pull request merged**